File: paper/src/main/java/ru/privatenull/pnauth/paper/PaperCommandSource.java
package ru.privatenull.pnauth.paper;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import ru.privatenull.pnauth.command.CommandSource;
import java.util.UUID;

/** Adapts Bukkit command senders to the shared command model. */
final class PaperCommandSource implements CommandSource {
    private final CommandSender sender;

    PaperCommandSource(CommandSender sender) { this.sender = sender; }
    @Override public UUID uniqueId() { return sender instanceof Player player ? player.getUniqueId() : null; }
    @Override public String username() { return sender instanceof Player player ? player.getName() : null; }
    @Override public boolean isPlayer() { return sender instanceof Player; }
    @Override public boolean hasPermission(String permission) { return sender.hasPermission(permission); }
}

File: paper/src/main/kotlin/ru/privatenull/pnauth/paper/PaperCommandSource.kt
package ru.privatenull.pnauth.paper

import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import ru.privatenull.pnauth.command.CommandSource
import java.util.UUID

/** Adapts Bukkit command senders to the shared command model. */
internal class PaperCommandSource(private val sender: CommandSender) : CommandSource {
    override fun uniqueId(): UUID? = if (sender is Player) sender.uniqueId else null
    override fun username(): String? = if (sender is Player) sender.name else null
    override fun isPlayer(): Boolean = sender is Player
    override fun hasPermission(permission: String): Boolean = sender.hasPermission(permission)
}
